lib/LongVectorLoweringPass.cpp

```diff
@@ -503,6 +521,35 @@ Value *LongVectorLoweringPass::visitCallInst(CallInst &I) {
   return V;
 }
 
+Value *LongVectorLoweringPass::visitCastInst(CastInst &I) {
+  auto *OriginalValue = I.getOperand(0);
+  auto *EquivalentValue = visitOrSelf(OriginalValue);
+  auto *OriginalDestTy = I.getDestTy();
+  auto *EquivalentDestTy = getEquivalentTypeOrSelf(OriginalDestTy);
+
+  // We expect something to lower, or this function shouldn't have been called.
+  assert(((OriginalValue != EquivalentValue) ||
+          (OriginalDestTy != EquivalentDestTy)) &&
+         "nothing to lower");
+
+  Value *V = nullptr;
+  switch (I.getOpcode()) {
+  case Instruction::BitCast: {
+    IRBuilder<> B(&I);
+    V = B.CreateBitCast(EquivalentValue, EquivalentDestTy, I.getName());
```

Comment:
This doesn't seem correct. You can't bitcast to/from aggregate types. So `bitcast <4 x i32> %in to <8 x i16>` would be invalid and need to be broken down to element-wise bitcasts and reconstructing the output aggregate.

Reply:
Indeed, you're correct. I missed the fact that bitcast apply only to _non-aggregate_ first class types. I've addressed this by keeping the scope to pointers for now and introduced a new tests with some TODOs for later.

Mind the fact that other instructions (alloca, load, store) do not have a dedicated explicit test. They are covered implicitely by the OpenCL vload/vstore tests. Let me know if you feel they deserve dedicated tests in LLVM IR format.

Reply:
I think it would be best if we had explicit coverage for each instruction in pass test.

Reply:
I've added a testfile for these three instructions and addressed an issue with alignment (that wasn't handled).